internal/mage/util/engine.go

```diff
@@ -202,10 +210,11 @@ func devEngineContainer(c *dagger.Client, arch string, version string, opts ...D
 	if err != nil {
 		panic(err)
 	}
-	return c.Container(dagger.ContainerOpts{Platform: dagger.Platform("linux/" + arch)}).
+
+	container := c.Container(dagger.ContainerOpts{Platform: dagger.Platform("linux/" + arch)}).
```

Comment:
Since we know that this image only works on `amd64`, should we fail if `arch` differs?

Reply:
`devEngineContainer` should be good as is because it's used for the creation of the regular engine's container. We could add a `panic` in `devEngineContainerWithGPUSupport` if the arch isn't `amd64`.

Reply:
Added a `panic`.